spec/manuals_publisher/removing_content_spec.rb

```diff
@@ -0,0 +1,58 @@
+feature "Removing content on Manuals Publisher", manuals_publisher: true do
+  include ManualsHelpers
+
+  let(:title) { title_with_timestamp }
+  let(:section_title) { title_with_timestamp }
+  let(:section_slug) { section_title.downcase.tr(" ", "-").gsub(/[^\w-]/, '') }
+
+  scenario "Remove a section from a published manual" do
+    given_there_is_a_published_manual_with_sections
+    when_i_remove_a_section_and_ignore_the_500
+    and_i_publish_the_manual
+    then_the_removed_section_redirects_to_the_root_page
+  end
+
+  def given_there_is_a_published_manual_with_sections
+    create_manual
+    create_manual_section
+    create_manual_section(title: section_title)
+    and_i_publish_the_manual
+  end
+
+  def when_i_remove_a_section_and_ignore_the_500
+    click_link section_title
+    click_link "Withdraw section"
+
+    fill_in "Change note", with: "Test removal"
+    click_button "Yes"
+
+    visit @edit_manual_url
+  end
+
+  def and_i_publish_the_manual
+    publish_manual
+  end
+
+  def then_the_removed_section_redirects_to_the_root_page
+    url = find_link("View on website")[:href]
+    removed_section_url = [url, section_slug].join("/")
+
+    reload_url_until_status_code(url, 200)
+    reload_url_until_status_code(removed_section_url, 301, keep_retrying_while: [200])
+
+    visit removed_section_url
+
+    expect(current_url).to eq(url)
+    expect(page).to have_content(title)
+    expect(page).to_not have_content(section_title)
+    expect_url_matches_live_gov_uk
+    expect_rendering_application("manuals-frontend")
+  end
+
+  private
```

Comment:
Don't really think we need to introduce privates into this, dare say every method could/should be private but think it only complicates things.

Also wonder if you need the method below since it introduces a few extra lines for effectively 1 line

Reply:
Makes sense.